### PR TITLE
Fix erratic behavior with single container child jumping outputs

### LINKED ISSUE #2466: Window in a single-child tabbed container moves across displays
Output of `i3 --moreversion 2>&- || i3 --version`:

```
Binary i3 version:  4.12 (2016-03-06, branch "4.12") © 2009 Michael Stapelberg and contributors
Running i3 version: 4.12 (2016-03-06, branch "4.12") (pid 896)
Loaded i3 config: /home/simon/.i3/config (Last modified: Wed 21 Sep 2016 10:41:51 AM PDT, 4076 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
```

URL to a logfile as per http://i3wm.org/docs/debugging.html:

http://logs.i3wm.org/logs/5713050069893120.bz2

**What I did:**
1. open a new, empty workspace
2. mod+enter+enter to create two terminals, side by side
3. focus right-hand terminal, type something in it to make sure it is distinguishable from the other terminal
4. mod+v to create a new container
5. mod+w to switch it to tabbed mode
6. alt+shift+left to move the terminal out of the tabbed container

**What I saw:**

On a multi-monitor setup, the terminal moves to the right-hand side of the monitor to the left. On a single-monitor setup (or if the workspace is on the far-left monitor), the terminal does not move at all, and the tabbed container does not get destroyed. I have reproduced this on two machines, and with the default config.

**What I expected instead:**

I expected the tabbed container to be destroyed, leaving the terminal in the same place. I have done this in older versions of i3 to destroy single-child containers, and it does not seem to be working anymore.

### PR DESCRIPTION
**Pull request by CyberShadow**
This fixes a regression introduced in commit 4e88c10564ca5366c2578908f62ec56625a26718: when attempting to move the single child of a container in the direction of another output, i3 would move the window to the output, despite the window not being at the edge of its output, instead of moving it to its parent container.

The bug occurred because the check for moving containers across outputs with non-default workspace layouts (issue #1603) did not actually verify that the moved window lies at the edge of the workspace, despite what its comment said.

Fixes #2466.

### CODE REVIEW

**Comment by Airblader** [comment]
Thanks for the PR! This looks mostly good to me.

@stapelberg We should probably rebase this to also have the changes to the tests you made, right? Also, we should perhaps put this in the bugfix.

File: testcases/t/535-move-single-to-output.t [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+#
+# • http://onyxneon.com/books/modern_perl/modern_perl_a4.pdf
+#   (unless you are already familiar with Perl)
+#
+# Tests that windows inside containers with a single child do not jump
+# over other containers and erratically move across outputs.
+# Ticket: #2466
+# Bug still in: 4.14-63-g75d11820
+use List::Util qw(first);
+use i3test i3_autostart => 0;
```
Comment (stapelberg) [inline testcases/t/535-move-single-to-output.t]: Please use `i3_config` here, see https://github.com/i3/i3/pull/2945

Comment (CyberShadow) [reply testcases/t/535-move-single-to-output.t]: Done.

File: testcases/t/537-move-single-to-output.t [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+#   (or docs/ipc)
+#
+# • http://onyxneon.com/books/modern_perl/modern_perl_a4.pdf
+#   (unless you are already familiar with Perl)
+#
+# Tests that windows inside containers with a single child do not jump
+# over other containers and erratically move across outputs.
+# Ticket: #2466
+# Bug still in: 4.14-63-g75d11820
+use List::Util qw(first);
```
Comment (stapelberg) [inline testcases/t/537-move-single-to-output.t]: Remove this line, you’re not using `first` anywhere.

Comment (CyberShadow) [reply testcases/t/537-move-single-to-output.t]: Done.

Comment (CyberShadow) [reply testcases/t/537-move-single-to-output.t]: Also removed unused `open_window` arguments.

**Pull request merged**